pandas/core/categorical.py

```diff
@@ -510,6 +512,58 @@ def base(self):
         return None
 
     @classmethod
+    def _from_inferred_categories(cls, inferred_categories, inferred_codes,
+                                  dtype):
+        """Construct a Categorical from inferred values
+
+        For inferred categories (`dtype` is None) the categories are sorted.
+        For explicit `dtype`, the `inferred_categories` are cast to the
+        appropriate type.
+
+        Parameters
+        ----------
+
+        inferred_categories, inferred_codes : Index
+        dtype : CategoricalDtype
+
+        Returns
+        -------
+        Categorical
+        """
+        from pandas import Index, to_numeric, to_datetime, to_timedelta
+
+        cats = Index(inferred_categories)
+
+        # Convert to a specialzed type with `dtype` is specified
+        if (isinstance(dtype, CategoricalDtype) and
```

Comment:
dtype by definition is already a CDT

Reply:
It could also be the string '`category'`. I've clarified the docstring.